From e9618e517a531b9a84cbcc408800904e76f5e39f Mon Sep 17 00:00:00 2001
From: Matt Revelle <matt@dynamictyping.org>
Date: Sun, 29 Nov 2009 14:46:16 -0500
Subject: [PATCH] Fixed #162: gen-class fails to accept type hints for arrays of primitives.  Rich's proposed fix works fine.

---
 src/clj/clojure/genclass.clj |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/clj/clojure/genclass.clj b/src/clj/clojure/genclass.clj
index 4947b3b..6f762b1 100644
--- a/src/clj/clojure/genclass.clj
+++ b/src/clj/clojure/genclass.clj
@@ -86,7 +86,7 @@
    (contains? prim->class x) (prim->class x)
    :else (let [strx (str x)]
            (clojure.lang.RT/classForName 
-            (if (some #{\.} strx)
+            (if (some #{\. \[} strx)
               strx
               (str "java.lang." strx))))))
 
-- 
1.6.4.2

